recipes/nmslib/all/conanfile.py

```diff
@@ -0,0 +1,67 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class Nmslib(ConanFile):
+    name = "nmslib"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nmslib/nmslib"
+    description = "Non-Metric Space Library (NMSLIB): An efficient similarity search library and a toolkit for evaluation of k-NN methods for generic non-metric spaces."
+    topics = ("knn-search", "non-metric", "neighborhood-graphs", "neighborhood-graphs", "vp-tree")
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+
+    _cmake = None
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.compiler.version == "14":
+                raise ConanInvalidConfiguration("Visual Studio 14 builds are not supported")  # TODO: add reason in message
+            if self.options.shared:
+                raise ConanInvalidConfiguration("Visual Studio shared builds are not supported (.lib artifacts missing)")
```

Comment:
I learnt recently that visual studio will not great the lib when there are no symbols exported

Reply:
yes, you're right